src/examples/3.test.ts

```diff
@@ -0,0 +1,30 @@
+import { totalDinnerExpenses, DailyFoodExpenses } from "./3";
+
+/**
+ * BUG REPORT:
+ * Test 3a passes, but test 3b fails.
+ */
+const expenses: DailyFoodExpenses[] = [
+  {
+    breakfast: 0,
+    lunch: 5,
+    dinner: 10,
+  },
+  {
+    breakfast: 10,
+    lunch: 5,
+    dinner: 0,
+  },
+];
+
+describe("3", () => {
+  // This test passes
+  test("3a", () => {
+    expect(totalDinnerExpenses(expenses, 0)).toStrictEqual(10);
```

Comment:
This example deviates substantially from the version on StackOverflow. The version there will return a NaN on *all* inputs. The other of this example promoted a local variable in that example to an optional parameter here. I'm not sure that really represents the problem on StackOverflow.

Reply:
This is one of the challenges with SO: the original snippet is incomplete and does not execute; e.g., the input is empty, collect() is never called, and the input structure is not provided -- except we know of the dinner/cena field.  

Here, I had to fill in the missing details so it would run.  I've removed the lunch and breakfast fields I added as they are not used.  If total is definitely never initialized, the IDE unambiguously labels it as a use before initialize mistake.  If we make that change, this becomes the most trivial bug, which may not be a bad thing.